objective.py

```diff
@@ -21,14 +21,14 @@ class Objective(BaseObjective):
 
     is_convex = False
 
-    requirements = ["scikit-learn"]
+    requirements = ["pip:scikit-learn"]
```

Comment:
As `scikit-learn` is a requirement for the `objective.py`, it is not necessary to add it to all the datasets and solvers.

Reply:
Indeed but I thought that it would be useful to add it to solvers and datasets so if the objective is changed the solvers they would still work